### PR TITLE
Big changes - SSM, proper cleanup, S3 publishing reports

### PR DESCRIPTION
**Pull request by aochsner**
Use build artifacts S3 for report publishing
Create SSM for build GithubToken
Proper cleanup of ECR & S3 buckets for deletion

BKIT-23

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,18 +44,57 @@ deps:
...
+
+update-deps:
+	@echo "Update SSM build parameters: /${OWNER}/${PROJECT}/build"
+	@read -p 'GitHub OAuth Token: ' REPO_TOKEN; \
+		aws cloudformation update-stack --stack-name "${OWNER}-${PROJECT}-deps" \
+                --region ${REGION} \
+			--template-body "file://cloudformation/deps/deps.yaml" \
+			--capabilities CAPABILITY_NAMED_IAM \
+			--parameters \
+				"ParameterKey=ParameterStoreNamespace,ParameterValue=/aochsner/bookit" \
```
Comment (defpearlpilot) [inline Makefile]: This should be changed to the owner as referenced below?

File: create-standard-riglet.sh [hunk 2]
```diff
@@ -10,12 +10,13 @@ read -p "Are you sure you want to proceed?  " -n 1 -r
...
 if [[ $REPLY =~ ^[Yy]$ ]]
 then
+  make create-deps && \
   echo "Y" | make create-foundation ENV=integration && \
   echo "Y" | make create-compute ENV=integration && \
   echo "Y" | make create-foundation ENV=staging && \
   echo "Y" | make create-compute ENV=staging && \
   echo "Y" | make create-foundation ENV=production && \
   echo "Y" | make create-compute ENV=production && \
-  echo "Y" | make create-build REPO=bookit-api REPO_BRANCH=master CONTAINER_PORT=8080 LISTENER_RULE_PRIORITY=100 && \
```
Comment (defpearlpilot) [inline create-standard-riglet.sh]: I couldn't really follow where the branch is specified going forward.

File: Makefile (hunk 1 above)
Comment (aochsner) [reply Makefile]: Yep absolutely.  Thanks!

File: create-standard-riglet.sh (hunk 2 above)
Comment (aochsner) [reply create-standard-riglet.sh]: So, the branch still defaults to `master` (line 10 of Makefile).  Can override that in the same way.

**Pull request merged**